src/ucm/event/event.c

```diff
@@ -123,23 +122,12 @@ static void ucm_event_dispatch(ucm_event_type_t event_type, ucm_event_t *event)
 
 static void ucm_event_enter()
 {
-    pthread_t self;
-
-    self = pthread_self();
-    if (pthread_mutex_trylock(&ucm_event_lock) != 0) {
-        ucs_memory_cpu_load_fence();
-        if (ucm_event_lock_owner == self) {
-            ucm_fatal("Detected recursive call to memory event handler");
-        }
-        pthread_mutex_lock(&ucm_event_lock);
-    }
-    ucm_event_lock_owner = self;
+    pthread_rwlock_rdlock(&ucm_event_lock);
```

Comment:
Should not we check for return value? (to make sure we do not own the lock already, also max num of simultaneous read lock may be exceeded)